File: GPG.Multiplayer.Statistics/GPG/Multiplayer/Statistics/PlayerAvatar.vb
Imports GPG.DataAccess
Imports GPG.Multiplayer.Quazal
Imports System

Namespace GPG.Multiplayer.Statistics
    Public Class PlayerAvatar
        Inherits MappedObject
        ' Methods
        Private Sub New()
        End Sub

        Public Sub New(ByVal record As DataRecord)
            MyBase.New(record)
        End Sub

        Public Shared Function GetPlayerAvatars(ByVal playerId As Integer) As PlayerAvatar()
            Dim item As New PlayerAvatar
            item.mPlayerID = playerId
            item.mAvatarID = 0
            Dim objects As MappedObjectList(Of PlayerAvatar) = New QuazalQuery("GetPlayerAvatarsByID", New Object() { playerId }).GetObjects(Of PlayerAvatar)
            objects.Insert(0, item)
            Return objects.ToArray
        End Function


        ' Properties
        Public ReadOnly Property Avatar As Avatar
            Get
                If (Me.mAvatarID <= 0) Then
                    Return Avatar.Default
                End If
                Return Avatar.AllAvatars.Item(Me.mAvatarID)
            End Get
        End Property

        Public ReadOnly Property ID As Integer
            Get
                Return Me.mID
            End Get
        End Property

        Public ReadOnly Property ManualAssignment As Boolean
            Get
                Return Me.mManualAssignment
            End Get
        End Property

        Public ReadOnly Property PlayerID As Integer
            Get
                Return Me.mPlayerID
            End Get
        End Property


        ' Fields
        <FieldMap("avatar_id")> _
        Private mAvatarID As Integer
        <FieldMap("player_avatar_id")> _
        Private mID As Integer
        <FieldMap("manual_assignment")> _
        Private mManualAssignment As Boolean
        <FieldMap("player_id")> _
        Private mPlayerID As Integer
    End Class
End Namespace

File: trunk/GPG.Multiplayer.Statistics/GPG/Multiplayer/Statistics/PlayerAvatar.cs
namespace GPG.Multiplayer.Statistics
{
    using GPG.DataAccess;
    using GPG.Multiplayer.Quazal;
    using System;

    public class PlayerAvatar : MappedObject
    {
        [FieldMap("avatar_id")]
        private int mAvatarID;
        [FieldMap("player_avatar_id")]
        private int mID;
        [FieldMap("manual_assignment")]
        private bool mManualAssignment;
        [FieldMap("player_id")]
        private int mPlayerID;

        private PlayerAvatar()
        {
        }

        public PlayerAvatar(DataRecord record) : base(record)
        {
        }

        public static PlayerAvatar[] GetPlayerAvatars(int playerId)
        {
            PlayerAvatar item = new PlayerAvatar();
            item.mPlayerID = playerId;
            item.mAvatarID = 0;
            MappedObjectList<PlayerAvatar> objects = new QuazalQuery("GetPlayerAvatarsByID", new object[] { playerId }).GetObjects<PlayerAvatar>();
            objects.Insert(0, item);
            return objects.ToArray();
        }

        public GPG.Multiplayer.Statistics.Avatar Avatar
        {
            get
            {
                if (this.mAvatarID <= 0)
                {
                    return GPG.Multiplayer.Statistics.Avatar.Default;
                }
                return GPG.Multiplayer.Statistics.Avatar.AllAvatars[this.mAvatarID];
            }
        }

        public int ID
        {
            get
            {
                return this.mID;
            }
        }

        public bool ManualAssignment
        {
            get
            {
                return this.mManualAssignment;
            }
        }

        public int PlayerID
        {
            get
            {
                return this.mPlayerID;
            }
        }
    }
}

